src/test-apps/happy/lib/WeaveState.py

```diff
@@ -168,6 +168,30 @@ def getNodeWeaveIPAddress(self, node_id=None, state=None):
                 return addr
         return None
 
+    def getNodeInterfaceWeaveIPAddress(self, interface_id, node_id, state=None):
```

Comment:
How does this control / influence whether that address is IPv4 or IPv6. Ostensibly that address should *only* be IPv6; however, the documentation should be clear about that fact and the code should, potentially, have an assertion or some such to trap / catch an IPv4 address that might "bubble up" through to this interface.

Reply:
Hi Grant:
Do you mean add a check to make sure there is ipv6 address in the node's address list before we do IP.prefixMatchAddress(weave_global_prefix, addr) in line 191?

This function will return weave ip address, which should be ipv6 address by default from my understanding.